Add and update various docs (FAQ, CHANGELOGs, READMEs, ...) for Jib plugins 3.0 release (#3189)
GoogleContainerTools/jib: docs/faq.md

A Dockerfile that performs a Jib-like build is shown below:

```Dockerfile

> FROM adoptopenjdk:11-jre

Reviewer: let's keep the comment about the default image

#### My shell script won't run
 

> The default base image used by the old Jib Maven and Gradle plugins did not have a shell, and thus shell scripts won't launch. See [Where is bash?](#where-is-bash) for more details.

Reviewer: ```suggestion
The default base image used by the Jib Maven and Gradle plugins prior to 3.0 did not have a shell, and thus shell scripts won't launch. See [Where is bash?](#where-is-bash) for more details.
```

#### My shell script won't run
 

> The default base image used by the Jib Maven and Gradle plugins prior to 3.0 did not have a shell, and thus shell scripts won't launch. See [Where is bash?](#where-is-bash) for more details.

Reviewer: maybe something like this?
```suggestion
Jib Maven and Gradle plugins prior to 3.0 used Distroless Java as the default base image, which does not have a shell. See [Where is bash?](#where-is-bash) for more details.
```

Reviewer: Actually, do we need this FAQ if we have the "Where is bash?" one

Author: We can take it out maybe a year later. It'll take a lot of time before the majority of people upgrade to 3.0, so I think it's possible that they'll hit this issue.
